### PR TITLE
Incident Finding event class definition

### PR DESCRIPTION
**Pull request by floydtree**
#### Description of changes:

1. Refactoring, @maxhotta 's PR to account for the new finding_info object, removing duplicates, adding descriptions.
2. Removing separate enum definitions, since that concept has been deprecated in OCSF.

A few topics to discuss based on https://github.com/ocsf/ocsf-schema/pull/786 - 

1. `findings_info` vs `finding_info_list` - I am okay with either of the two, however finding_info_list makes it easier to distinguish it from finding_info. - based on weekly call - `finding_info_list`
2. `user` vs `assignee` - are both necessary in the class? - keeping assignee, adding `assignee_group`
3. `incident_uid` - haven't added it, `metadata_uid` should be used instead.

### CODE REVIEW

**Comment by pagbabian-splunk** [comment]
Generally looks good and needed for complettion.

Main comment so far: `assignee` could also be a `group` of users; typically in a SOC with shifts, incidents could be passed across shifts to users in a group.

**Review by mikeradka — DISMISSED** [review]

**Comment by maxhotta** [comment]
Looks good to me as well.  The only attribute that I noticed missing was the consolidated attacks array.  I suspect it was removed for duplication as it's already a part of each of the findings?

**Comment by pagbabian-splunk** [comment]
> Looks good to me as well. The only attribute that I noticed missing was the consolidated attacks array. I suspect it was removed for duplication as it's already a part of each of the findings?

That's what I would assume as well - the `finding_info_list` has `finding_info.attacks` for each Finding in the Incident.

**Comment by floydtree** [comment]
>Looks good to me as well. The only attribute that I noticed missing was the consolidated attacks array. I suspect it was removed for duplication as it's already a part of each of the findings?

> That's what I would assume as well - the finding_info_list has finding_info.attacks for each Finding in the Incident.

Yep, that was the thought, but we can add a top level `attacks` as Paul suggested for an incident's summary of ttps.

**Review by zschmerber — APPROVED** [review]
LGTM

**Review by mikeradka — APPROVED** [review]

File: dictionary.json [hunk 1]
```diff
@@ -178,6 +178,16 @@
       "description": "The arguments sent along with the HTTP request.",
       "type": "string_t"
     },
+    "assignee": {
+      "caption": "Assignee",
+      "description": "The details of the user assgined to an Incident.",
```
Comment (Aniak5) [inline dictionary.json:183]: Misspelled `assgined`

File: dictionary.json [hunk 2]
```diff
@@ -178,6 +178,16 @@
...
+      "type": "user"
+    },
+    "assignee_group": {
+      "caption": "Assignee Group",
+      "description": "The details of the group assgined to an Incident.",
```
Comment (Aniak5) [inline dictionary.json:188]: Misspelled `assgined`

File: dictionary.json (hunk 1 above)
Comment (floydtree) [reply dictionary.json]: good catch! fixed

File: dictionary.json [hunk 3]
```diff
@@ -2530,13 +2546,38 @@
...
+        "0": {
+          "description": "No priority is assigned.",
+          "caption": "Unknown"
+        },
+        "1": {
+          "description": "Application or personal procedure unusable, where a workaround is available or a repair is possible.",
+          "caption": "Low"
+        },
+        "2": {
+          "description": "Non-critical function or procedure, unusable or hard to use having an operational impact, but with no direct impact on services availability. A workaround is available.",
```
Comment (Aniak5) [inline dictionary.json:2565]: This is a little confusing to read. Maybe something like "Non-critical function or procedure is unusable or hard to use, causing operational disruptions with no direct impact on services availability."

File: dictionary.json [hunk 4]
```diff
@@ -3309,6 +3387,11 @@
       "description": "The last or family name for the user.",
       "type": "string_t"
     },
+    "is_suspected_breach": {
+      "caption": "Suspected Breach",
+      "description": "A determination based on analytics whether a potential breach was found.",
```
Comment (Aniak5) [inline dictionary.json:3392]: Might be a bit more clear as: "A determination based on analytics as to whether a potential breach was found."

File: events/findings/incident_finding.json [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    "assignee": {
+      "group": "context",
+      "requirement": "optional"
+    },
+    "assignee_group": {
+      "group": "context",
+      "requirement": "optional"
+    },
+    "attacks": {
+      "description": "An array of <a target='_blank' href='https://attack.mitre.org'>MITRE ATT&CK®</a> objects describing the tactics, techniques & sub-techniques assocaited to the Incident.",
```
Comment (Aniak5) [inline events/findings/incident_finding.json:40]: Misspelled `assocaited`

File: events/findings/incident_finding.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+      "group": "primary",
+      "requirement": "required"
+    },
+    "is_suspected_breach": {
+      "group": "context",
+      "requirement": "optional"
+    }
+  },
+  "constraints": {
+    "at_least_one" :[
```
Comment (Aniak5) [inline events/findings/incident_finding.json:149]: Nit: Space between semicolon and `"at_least_one"`

**Review by Aniak5 — APPROVED** [review]
LGTM!

**Review by zschmerber — APPROVED** [review]

**Review by maxhotta — APPROVED** [review]

**Pull request merged**